music/src/core/recorder.ts

```diff
@@ -315,14 +333,21 @@ export class Recorder {
     const cmd = event.data[0] >> 4;
     const pitch = event.data[1];
     const velocity = (event.data.length > 2) ? event.data[2] : 1;
+    const device = event.srcElement;
 
     // Some MIDI controllers don't send a separate NOTE_OFF command.
     if (cmd === NOTE_OFF || (cmd === NOTE_ON && velocity === 0)) {
+      if (this.callbackObject) {
```

Comment:
`callbackObject` might be defined, but `noteOn`/`noteOff` might not be (outside of typescript nobody will yell at you), so I would check if those exist as well.